ludwig/trainers/trainer.py

```diff
@@ -1192,6 +1185,13 @@ def _train_loop(
                 if should_break:
                     return should_break
```

Comment:
Will we be one epoch off if we return here?

Reply:
I think when the end of an epoch lines up with eval and early stopping, the final epoch increment won't happen.